modules/extensions-support/src/main/java/org/mule/module/extension/internal/manager/ExtensionRegister.java

```diff
@@ -0,0 +1,241 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.module.extension.internal.manager;
+
+import org.mule.extension.exception.NoSuchExtensionException;
+import org.mule.extension.introspection.Configuration;
+import org.mule.extension.introspection.Extension;
+import org.mule.extension.introspection.Operation;
+import org.mule.extension.runtime.ConfigurationInstanceProvider;
+import org.mule.util.CollectionUtils;
+
+import com.google.common.cache.CacheBuilder;
+import com.google.common.cache.CacheLoader;
+import com.google.common.cache.LoadingCache;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableSet;
+
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+
+import org.apache.commons.collections.Predicate;
+
+/**
+ * A utility class to hold the state related to registered
```

Comment:
A utility class to hold the state related to registered > Holds the state of registered.
realising instances > instances